src/commands/docs/migrate.ts

```diff
@@ -0,0 +1,189 @@
+
+  static hidden = true;
+
+  static summary = `Migrates a directory of pages to the ReadMe Guides format.\n\n${alphaNotice}`;
+
+  static description =
+    "The path can either be a directory or a single Markdown file. The command will transform the Markdown using plugins and validate the frontmatter to conform to ReadMe's standards.";
+
+  static args = {
+    path: Args.string({ description: 'Path to a local Markdown file or folder of Markdown files.', required: true }),
+  };
+
+  static flags = {
+    out: Flags.string({
+      summary: 'The directory to write the migration output to. Defaults to a temporary directory.',
+    }),
+    'skip-validation': Flags.boolean({
+      description:
+        'Skips the validation of the Markdown files. Useful if this command is as part of a chain of commands that includes `docs upload`.',
+    }),
+  };
+
+  async run() {
+    const { path: rawPathInput }: { path: string } = this.args;
+    const { out: rawOutputDir, 'skip-validation': skipValidation } = this.flags;
+
+    const outputDir = rawOutputDir || (await dir({ prefix: 'rdme-migration-output' })).path;
+
+    let pathInput = rawPathInput;
+
+    // todo: fix this type once https://github.com/oclif/core/pull/1359 is merged
+    const fileScanHookResults: Hook.Result<PluginHooks['pre_markdown_file_scan']['return']> = await this.config.runHook(
+      'pre_markdown_file_scan',
+      { pathInput },
+    );
+
+    fileScanHookResults.successes.forEach(success => {
+      if (success.result) {
+        pathInput = success.result;
+      }
+    });
+
+    fileScanHookResults.failures.forEach(fail => {
+      if (fail.error && fail.error instanceof Error) {
+        throw new Error(`Error executing the \`${fail.plugin.name}\` plugin: ${fail.error.message}`);
+      }
+    });
+
+    let unsortedFiles = await findPages.call(this, pathInput);
+
+    let transformedByHooks = false;
+
+    // todo: fix this type once https://github.com/oclif/core/pull/1359 is merged
+    const validationHookResults: Hook.Result<PluginHooks['pre_markdown_validation']['return']> =
+      await this.config.runHook('pre_markdown_validation', { pages: unsortedFiles });
+
+    if (!validationHookResults.successes.length && !validationHookResults.failures.length) {
+      throw new Error('This command requires a valid migration plugin.');
+    }
+
+    validationHookResults.successes.forEach(success => {
+      if (success.result && success.result.length) {
+        transformedByHooks = true;
+        this.log(`🔌 ${success.result.length} Markdown files updated via the \`${success.plugin.name}\` plugin`);
+        unsortedFiles = success.result;
+      }
+    });
+
+    validationHookResults.failures.forEach(fail => {
+      if (fail.error && fail.error instanceof Error) {
+        throw new Error(`Error executing the \`${fail.plugin.name}\` plugin: ${fail.error.message}`);
+      }
+    });
+
+    // todo: either DRY this validation logic up or remove it entirely
```

Comment:
mostly everything from this point onward is ripped from the `syncPagePath` file.